classes/common.php

```diff
@@ -4,16 +4,29 @@
 defined('MOODLE_INTERNAL') || die();
 
 class common {
-    public static function reset_entry($conditions){
+    public static function reset_entry($conditions, $force = false){
         global $DB;
+
         $oldentry = $DB->get_record('availability_examus', $conditions);
-        if ($oldentry and $oldentry->status != 'Not inited') {
+
+        $not_inited = $oldentry && $oldentry->status == 'Not inited';
+
```

Comment:
Mixing `&&` and `||` in one condition is quiet hard to understand.
I suggest to split and reverse the condition:
```suggestion
if (!$oldentry) return false
else if ($not_inited && !$force) return false
else {
```